cpp/src/arrow/flight/transport/grpc/serialization_internal.cc

```diff
@@ -380,6 +382,14 @@ ::grpc::Status FlightDataDeserialize(ByteBuffer* buffer,
           return ::grpc::Status(::grpc::StatusCode::INTERNAL,
                                 "Unable to read FlightData body");
         }
+        // XXX: due to where we sit, we can't use a custom allocator
+        // XXX: any error here will likely crash or hang gRPC!
+        auto status =
+            util::EnsureAlignment(std::move(out->body), 64, default_memory_pool())
```

Comment:
Isn't 64 a bit heavy-handed? Basically, and assuming the distribution of gRPC slice alignments is uniform, we will reallocate almost all incoming buffers?

Reply:
This should be configurable as not all code using the C++ flight client is sensitive to alignment.